src/main/scopes/jsx/jsx-scope.ts

```diff
@@ -58,28 +59,67 @@ export class JsxScope extends Scope {
    * directory's project.
    */
   @Trace()
-  async captureElementMetrics(): Promise<void> {
+  async captureAllMetrics(): Promise<void> {
     const importMatchers = [
       new AllImportMatcher(),
       new NamedImportMatcher(),
       new RenamedImportMatcher()
     ]
-    const instrumentedPackage = await getPackageData(this.cwd, this.logger)
-    const sourceFiles = await getTrackedSourceFiles(this.root, this.logger)
+    const instrumentedPackage = await getPackageData(this.cwd, this.cwd, this.logger)
+    const sourceFiles = await this.findRelevantSourceFiles(instrumentedPackage)
+
+    this.logger.debug('Filtered source files: ' + sourceFiles.map((f) => f.fileName))
 
     const promises: Promise<void>[] = []
 
     for (const sourceFile of sourceFiles) {
+      const resultPromise = this.captureFileMetrics(
+        sourceFile,
+        instrumentedPackage.name,
+        importMatchers
+      )
+
       if (this.runSync) {
-        await this.captureFileMetrics(sourceFile, instrumentedPackage.name, importMatchers)
+        await resultPromise
       } else {
-        promises.push(this.captureFileMetrics(sourceFile, instrumentedPackage.name, importMatchers))
+        promises.push(resultPromise)
       }
     }
 
     await Promise.allSettled(promises)
   }
 
+  /**
+   * Finds tracked source files and then filters them based on ones that appear in a  project which
+   * depends on the in-context instrumented package/version.
+   *
+   * @param instrumentedPackage - Data about the instrumented package to use during filtering.
+   * @returns A (possibly empty) array of source files.
+   */
+  async findRelevantSourceFiles(instrumentedPackage: PackageData) {
+    const sourceFiles = await getTrackedSourceFiles(this.root, this.logger)
+
+    const filterPromises = sourceFiles.map(async (f) => {
+      const prefix = await getDirectoryPrefix(path.dirname(f.fileName), this.logger)
+      const prefixPackageData = await getPackageData(prefix, this.root, this.logger)
+      const installingPackages = await findInstallingPackages(
+        this.cwd,
+        this.root,
+        instrumentedPackage,
+        this.logger
+      )
```

Comment:
could this be outside of the map? so it isn't recomputed on every iteration?

Reply:
Definitely! Good spot! Since this is now cached, it shouldn't slow things down to have it in the map, but I'll move it out since that's definitely how it should be.